chain/chain/src/test_utils.rs

```diff
@@ -48,13 +50,19 @@ pub fn get_chain_with_epoch_length_and_num_shards(
     num_shards: NumShards,
 ) -> Chain {
     let store = create_test_store();
-    let chain_genesis = ChainGenesis::new(&GenesisConfig::test());
-    let vs = ValidatorSchedule::new()
-        .block_producers_per_epoch(vec![vec!["test1".parse().unwrap()]])
-        .num_shards(num_shards);
-    let epoch_manager = MockEpochManager::new_with_validators(store.clone(), vs, epoch_length);
+    let mut genesis = Genesis::test_sharded(
+        vec!["test1".parse::<AccountId>().unwrap()],
+        1,
+        vec![1; num_shards as usize],
+    );
+    genesis.config.epoch_length = epoch_length;
+    let tempdir = tempfile::tempdir().unwrap();
```

Comment:
Would it be possible to somehow make this module available in test contexts only, so that the `tempfile` dependency didn't have to move to regular dependencies? Not super important, perfectly fine with me if you don't want or have time to investigate.

Reply:
Good point, I will take a look. I'm merging this PR as it is and I will create a separate PR for making `test_utils` only usable in test context.

Reply:
SGTM

Reply:
So I've tried adding `#[cfg(test)]` for test_utils but that doesn't work because this module is exported and used in other crates. What would be a good way to deal with that? I've found a solution with a separate feature [here](https://github.com/rust-lang/cargo/issues/8379), do you think that is a good idea?
cc @Ekleog-NEAR

Reply:
Adding a separate feature flag is a good way to go about this, but it won't help much in our specific case at all -- so long as there's any crate in our workspace enabling the feature, it will get enabled everywhere regardless.

No point to spin our wheels on it right now, then. It sounds like something that we could eventually see our test loop framework incorporate in some way perhaps.